src/background/index.ts

```diff
@@ -46,23 +44,40 @@ browser.webRequest.onBeforeSendHeaders.addListener(
 )
 
 // Update on tab change or new urls
-browser.tabs.onActivated.addListener(() => getCurrentLocale())
-browser.tabs.onUpdated.addListener(() => getCurrentLocale())
+browser.tabs.onActivated.addListener((tabId) => setBadge(tabId.tabId))
+browser.tabs.onUpdated.addListener(async () => {
+  const tabId = await getCurrentTabId()
+  if (tabId) setBadge(tabId)
+})
+
+function sendCurrentState(tabId: number) {
+  const locale: Locale = (tabId && ram.get(tabId)) || null
+  const msg: MessageType = { type: 'current-output', data: locale }
+  browser.runtime.sendMessage(msg)
+}
 
 // Listener for messages from content scripts
-browser.runtime.onMessage.addListener(({ type, data }) => {
-  switch (type) {
-    case 'get':
-      updateLocale(data)
+browser.runtime.onMessage.addListener(async (message: MessageType, sender) => {
+  switch (message.type) {
```

Comment:
Could we name these events differently? Right now they're very generic words and a bit hard to inuitively understand what each one does.

Suggestions:
- `sync` -> `setBackgroundLocaleFromTab`
- `update` -> `setBackgroundLocaleFromPopup`
- `current-input` -> `getBackgroundLocale`
- `current-output` -> `setPopupLocale`

Reply:
Yeah, was strugling with renaming them and then forgot to do that xD